### PR TITLE
Upgrade react native to 0.53.3

### LINKED ISSUE #3377: Upgrade to latest React Native
### User Story

As a developer, I want to rely on latest library versions

### Description

*Type*: Feature

Each new release of React Native brings fixes and new features. We don't want to lag behind too long.
Upgrade to React Native `0.53` (or `0.54` if available).

### PR DESCRIPTION
**Pull request by voxlet**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #3377

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)

Upgrades:
- React Native to 0.53.3
- Re-Natal to 0.8.1 (required for changes in metro-bundler -> metro)

### Review notes:

Includes workarounds for [react-native-vector-icons (or rather, Metro)](https://github.com/oblador/react-native-vector-icons/issues/626#issuecomment-362386341) and [FlatList keyExtractor](https://github.com/wonday/react-native-pdf/issues/125).

As a side note, we could probably safely upgrade to re-frame 0.10.2 -> 0.10.4 which is what [Re-Natal 0.8.1 is on](https://github.com/drapanjanas/re-natal/releases/tag/v0.8.1).

### Steps to test:

- `rm -rf node_modules`
- Rebuild according to [Wiki](https://wiki.status.im/Building_Status)
- Confirm nothing is obviously broken in app
- Confirm tests pass (with the exception of `group-transactions-by-date`, which also fails in develop, though I suspect the tests do not actually test React to begin with)

status: ready

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,14 +75,14 @@
...
+    "rn-snoopy": "github:status-im/rn-snoopy",
     "stream-browserify": "1.0.0",
     "string_decoder": "0.10.31",
     "timers-browserify": "1.4.2",
     "tty-browserify": "0.0.0",
     "url": "0.10.3",
     "vm-browserify": "0.0.4",
-    "web3": "github:status-im/web3.js#status-develop",
-    "rn-snoopy": "github:status-im/rn-snoopy"
+    "web3": "github:status-im/web3.js#status-develop"
```
Comment (jeluard) [inline package.json:84]: Can we use exact version match here?

**Comment by jeluard** [comment]
@voxlet Nice work! Just make sure we are locking RN version down.

Related to `keyExtractor` we also have https://github.com/status-im/status-react/issues/3064 : maybe you want to take a look at it too? :)

**Comment by voxlet** [comment]
@jeluard Fixed, thanks for the catch!

#3064 looks like something I could tackle as well. I'll put it on my list :)

File: env/dev/env/android/main.cljs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (ns ^:figwheel-no-load env.android.main
   (:require [reagent.core :as r]
             [status-im.android.core :as core]
-            [figwheel.client :as figwheel :include-macros true]
+            [figwheel.client :as fw]
```
Comment (flexsurfer) [inline env/dev/env/android/main.cljs]: we are using full names , `figwheel` was correct

File: env/dev/env/ios/main.cljs [hunk 3]
```diff
@@ -3,10 +3,9 @@
             [re-frisk-remote.core :as rr]
             [status-im.ios.core :as core]
             [status-im.utils.handlers :as utils.handlers]
-            [figwheel.client :as figwheel :include-macros true]
+            [figwheel.client :as fw]
```
Comment (flexsurfer) [inline env/dev/env/ios/main.cljs]: we are using full names , `figwheel` was correct

File: rn-cli.config.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// workaround for react-native-vector-icons
```
Comment (flexsurfer) [inline rn-cli.config.js:1]: could you elaborate why we need this?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: rn-cli.config.js (hunk 4 above)
Comment (voxlet) [reply rn-cli.config.js:1]: Apparently some changes in Metro's file resolving behavior surfaced packaging issues in react-native-vector-icons. This file works around this by ignoring the offending files.

A longer explanation [here](https://github.com/facebook/metro/issues/139#issuecomment-366213751).

**Comment by voxlet** [comment]
@flexsurfer Thanks for the feedback! Pushed fixes for the figwheel alias.

To provide some background, changes in both files were the result of `re-natal upgrade`. There may be some advantage to adhering to their naming conventions, as the next upgrade will likely revert our changes back. Then again, a clean upgrade patch will not happen anyway since we have already have custom code there.

Ideally, there would be some way to keep the code separate so re-natal can cleanly patch when upgrading - maybe something to look into in the future (not that it is too important, considering the frequency of upgrading).

**Comment by jeluard** [comment]
@voxlet What is `re-natal upgrade` doing? I'd like to move away from re-natal

File: rn-cli.config.js (hunk 4 above)
Comment (voxlet) [reply rn-cli.config.js:1]: I understand the files ignored are test fixtures, so there should be no harm in ignoring them in our builds.

**Comment by voxlet** [comment]
@jeluard Looks like it simply copies over the files in `env/` and `figwheel-bridge.js`, so any changes there would be blown away on `re-natal upgrade`.

```
#{devEnvRoot}/env/#{platform}/main.cljs
#{config.envRoots.dev}/env/config.cljs
#{devEnvRoot}/user.clj
#{prodEnvRoot}/env/#{platform}/main.cljs
./figwheel-bridge.js
```

https://github.com/drapanjanas/re-natal/blob/26b9a4e8f071e6ab49ddf97da8dca3c7fb3ce082/re-natal.coffee#L337-L368

https://github.com/drapanjanas/re-natal/blob/26b9a4e8f071e6ab49ddf97da8dca3c7fb3ce082/re-natal.coffee#L792

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/drapanjanas/re-natal">drapanjanas/re-natal</a></strong></div><div>re-natal - Bootstrap ClojureScript React Native apps</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/drapanjanas/re-natal">drapanjanas/re-natal</a></strong></div><div>re-natal - Bootstrap ClojureScript React Native apps</div></blockquote>

**Comment by jeluard** [comment]
Yes in our case it's probably useless. Our `figwheel-bridge.js` is already significantly different and platform files don't move much.

**Review by flexsurfer — APPROVED** [review]

**Comment by voxlet** [comment]
@jeluard Agreed, I ended up reverting most of `re-natal upgrade` in this case as well.

Digging around the re-natal script turned out to be good learning experience, though! Demystified a lot of what's going on behind the scenes, and what it actually takes to use CLJS on React Native.

**Comment by jeluard** [comment]
@voxlet Great! Now that you have this behind the scene knowledge you might be interested with https://github.com/status-im/status-react/issues/3398 ! Let me know what you think.

**Review by jeluard — APPROVED** [review]

**Comment by lukaszfryc** [comment]
I've tested the whole app and the only thing I found is that the "Type your password" and "Confirm" hints are not fully visible:



To reproduce: 
1. Go to Console to create a new account
2. Click on the password message to trigger `/password` command

@voxlet could you please check it?

**Review by lukaszfryc — DISMISSED** [review]
@voxlet there is a small issue with a hint in input fields. Please check my latest comment.

**Comment by status-open-bounty** [comment]
Contract deploying
Transaction: [0xa1caf6e51346aa04e4ca353c4f6a3b884ca5e8cdfe9fc5952262074d74864243](https://etherscan.io/tx/0xa1caf6e51346aa04e4ca353c4f6a3b884ca5e8cdfe9fc5952262074d74864243)

**Comment by voxlet** [comment]
@lukaszfryc Fixed, though this is probably more a workaround than a proper fix. `on-layout` is not getting called on for the text-input on Android, leaving `container-width` nil. I used the content width from `on-content-size-change` in its place.

**Comment by jeluard** [comment]
@voxlet Please resolve conflict

**Comment by voxlet** [comment]
@jeluard rebased to current develop

**Comment by jeluard** [comment]
@voxlet Also please squash your commits!

**Pull request merged**